src/rgw/librgw.cc

```diff
@@ -220,7 +220,7 @@ namespace rgw {
     rgw_env.set("HTTP_HOST", "");
 
     /* XXX and -then- bloat up req_state with string copies from it */
-    struct req_state rstate(req->cct, &rgw_env, req->get_user());
+    struct req_state rstate(req->cct, &rgw_env, req->get_user(), 0);
```

Comment:
having not read the change to the req_state ctor, this is obscure, I'd put a gloss in /* */ after 0

Reply:
do you think it would be better for the ctor to have a default value of 0, so the call sites could just omit the argument?

Reply:
either is fine